Task 2/2.1/2.1.2/Figures.cs

```diff
@@ -0,0 +1,169 @@
+            this.centre = centre;
+            this.radius = radius;
+        }
+
+        public override double GetLength()
+        {
+            return 2 * Math.PI * radius;
+        }
+
+        public override bool Equals(object argument)
+        {
+            if (argument == null || !(argument is Сircular)) return false;
+            Сircular newArgument = argument as Сircular;
+
+            if (this.centre == newArgument.centre && this.radius == newArgument.radius)
+            {
+                return true;
+            }
+            return false;
+        }
+
+        public override string ToString()
+        {
+            return name + " - { Center: " + centre + "; radius: " + radius + "; length:" + GetLength() + "}";
+        }
+    }
+
+    public class Round : Сircular //круг
+    {
+        public Round(Point centre, double radius) : base("Round", centre, radius) { }
+
+        public override double GetArea()
+        {
+            return Math.PI * Math.Pow(radius, 2);
+        }
+
+        public override string ToString()
+        {
+            return name + " - { Center: " + centre + "; radius: " + radius + "; length:" + GetLength() + "; area: " + GetArea() + "}";
+        }
+    }
+
+    public class Circle : Сircular //окружность
+    {
+        public Circle(Point centre, double radius) : base("Circle", centre, radius){}
+
+        public override double GetArea()
+        {
+            return 0;
+        }
+    }
+
+    public class Ring : Figure //кольцо
+    {
+        public Point centre { get; }
+        public Round outherCircle { get; }
+        public Round innerCircle { get; }
+
+        public Ring(Point centre, double outherRadius, double innerRadius) : base("Ring")
+        {
+            if (outherRadius <= innerRadius) throw new Exception("Outher radius cann't be < or = inner radius");
+            this.centre = centre;
+            outherCircle = new Round(centre, outherRadius);
+            innerCircle = new Round(centre, innerRadius);
+        }
+
+        public override double GetLength()
+        {
+            return outherCircle.GetLength() + innerCircle.GetLength();
+        }
+
+        public override double GetArea()
+        {
+            return outherCircle.GetArea() - innerCircle.GetArea();
+        }
+
+        public override bool Equals(object argument)
+        {
+            if (argument == null || !(argument is Ring)) return false;
+
+            Ring newArgument = argument as Ring;
+            if (this.centre == newArgument.centre && this.outherCircle == newArgument.outherCircle && this.innerCircle == newArgument.innerCircle)
+            {
+                return true;
+            }
+            return false;
+        }
+
+        public override string ToString()
+        {
+            return name + " - { Center: " + centre + "; outher radius: " + outherCircle.radius + "; inner radius: " + innerCircle.radius + "; length:" + GetLength() + "; area: " + GetArea() + "}";
```

Comment:
This line is too long. I don't say that line must fit in 80 lines, like it's written in old guides, but when it's more than 120 - it's harder to read. 150 - is my limit.